### PR TITLE
ISPN-13686 Feature form for transactional cache

### PR DESCRIPTION
**Pull request by dipguptaredhat**
Feature form for transactional cache

### CODE REVIEW

**Comment by karesti** [comment]
@dipguptaredhat did you check @pruivo update?

**Comment by dipguptaredhat** [comment]
> @dipguptaredhat did you check @pruivo update?

Will do, currently solving bugs for Backups cache.

**Comment by dipguptaredhat** [comment]
@karesti @pruivo do we need other attributes also?

**Comment by pruivo** [comment]
@dipguptaredhat 
https://infinispan.zulipchat.com/#narrow/stream/118645-infinispan/topic/Cache.20Wizard/near/283987185

**Comment by pruivo** [comment]
if you want to split them into "mandatory" and "advanced" I would do the following
* isolation-level, locking-mode, and tx-mode in "mandatory" (with default value set)
* the remaining attributes in advanced/tunning section

**Comment by pruivo** [comment]
@dipguptaredhat `isolation-level` only affects transactional caches. Should be part of transactional basic features.

**Comment by dipguptaredhat** [comment]
@karesti @pruivo pushed a new commit, adding more options and labels exported

**Comment by pruivo** [comment]
* `Read isolation` => should be in transactional capabilities. `REPEATABLE_READ` should be the default
* `Transaction mode` => `NONE` and `BATCH` does not make much sense in server mode and can be removed. `NON_XA` should be the default
* `Transaction manager lookup` => put default value?
* `Recovery cache` => put default value? Hide if `Transaction mode` is not `FULL_XA`
* `Notifications` => can be remove. I've just checked and cannot be used in server mode :( 

BTW, my 2cents, what about making `Read isolation` and `Transaction mode` as radio buttons instead of a drop-down menu?

**Comment by dipguptaredhat** [comment]
@pruivo `Transactional mode ` has 3 options, wouldn't it be better to have a drop-down select menu?

**Comment by dipguptaredhat** [comment]
@pruivo your suggested changes done :+1:

File: src/app/assets/languages/en.json [hunk 1]
```diff
@@ -264,7 +277,21 @@
...
+          "stop-timeout": "Stop timeout",
+          "stop-timeout-tooltip": "Sets the amount of time, in milliseconds, that the indexing thread waits for background merges to finish before it shuts down. If the indexing thread is still running after the timeout, it will be interrupted and the indexing process will stop.",
+          "transaction-manager-lookup": "Transaction manager lookup",
+          "transaction-manager-lookup-tooltip": "Sets the transaction manager lookup class. This class is responsible for managing transactions for the cache. The default implementation is org.infinispan.transaction.lookup.GenericTransactionManagerLookup.",
+          "complete-timeout": "Complete timeout",
+          "complete-timeout-tooltip": "Sets the amount of time, in milliseconds, that the transaction manager waits for a transaction to complete before it times out. If a transaction does not complete within the timeout, it is aborted.",
```
Comment (pruivo) [inline src/app/assets/languages/en.json:286]:
```suggestion
          "complete-timeout-tooltip": "Sets the duration, in milliseconds, in which {{brandname}} keeps the information about the completion of a transaction and aborts remote transactions that are running for more than this duration",
```

File: src/app/assets/languages/en.json [hunk 2]
```diff
@@ -264,7 +277,21 @@
...
+          "reaper-interval": "Reaper interval",
+          "reaper-interval-tooltip": "Sets the amount of time, in milliseconds, that a transaction reaper thread waits before checking for timed out transactions. If a transaction has not completed within the timeout, it is aborted.",
```
Comment (pruivo) [inline src/app/assets/languages/en.json:288]:
```suggestion
          "reaper-interval-tooltip": "Sets the interval, in milliseconds, that a transaction reaper thread checks if the transaction is not completed within the timeout.",
```

**Comment by oraNod** [comment]
@dipguptaredhat and @pruivo Hey guys, FYI I've been trying to get around to the strings on this PR. Can you give me a shout when you're ready with all the changes please? I can do a final round before merge. Thanks.

File: src/app/assets/languages/en.json [hunk 3]
```diff
@@ -214,7 +214,18 @@
           "index-storage-entity-helper": "Each entity that you specify must be unique.",
           "secured": "Security authorization",
           "secured-description": "Restrict access to the cache to protect data. {{brandname}} denies cache operations if users do not have one of the roles you select.",
-          "select-roles": "Select roles"
+          "select-roles": "Select roles",
+          "transactional": "Transactional",
+          "transactional-description": "Transactional tooltip",
+          "transactional-mode": "Transactional mode",
+          "transactional-mode-tooltip": "Configure the transactional mode for the cache. Transactional caches are not shared across nodes and are not replicated. Transactional caches are useful for high-performance transactional data stores.",
```
[the remark this answers is not in the document]
Comment (dipguptaredhat) [reply src/app/assets/languages/en.json:221]: no, it was my bad, wasn't copied from anywhere.

File: src/app/assets/languages/en.json [hunk 4]
```diff
@@ -214,7 +214,21 @@
...
           "secured": "Security authorization",
           "secured-description": "Restrict access to the cache to protect data. {{brandname}} denies cache operations if users do not have one of the roles you select.",
-          "select-roles": "Select roles"
+          "select-roles": "Select roles",
+          "transactional": "Transactional",
+          "transactional-description": "Allows {{brandname}} to participate in distributed transaction complient with JTA",
+          "transactional-mode": "Transactional mode",
+          "transactional-mode-tooltip": "Configure the transactional mode for the cache. Transactional caches are not shared across nodes and are not replicated. Transactional caches are useful for high-performance transactional data stores.",
+          "non-xa": "NON_XA",
+          "non-xa-tooltip": "NON_XA tooltip",
```
Comment (dipguptaredhat) [inline src/app/assets/languages/en.json:223]: > can we get a tooltip for each option or should we put all descriptions in `transactional-mode-tooltip`?
extra tooltip added for radio options, you can add here @pruivo

**Comment by dipguptaredhat** [comment]
@pruivo branch rebased :+1:

File: src/app/assets/languages/en.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (pruivo) [reply src/app/assets/languages/en.json:272]: can you remove it?

File: src/app/assets/languages/en.json [hunk 5]
```diff
@@ -265,7 +265,21 @@
...
           "mode-tooltip": "Control whether state transfer happens manually on user action, which is the default, or automatically when backup locations come online. {{brandname}} can perform automatic state transfer with the ASYNC backup strategy only.",
           "mode-manual": "MANUAL",
-          "mode-auto": "AUTOMATIC"
+          "mode-auto": "AUTOMATIC",
+          "transactional": "Transactional",
+          "transactional-description": "Allows {{brandname}} to participate in distributed transaction complient with JTA",
+          "transactional-mode": "Transactional mode",
+          "transactional-mode-tooltip": "Configure the transactional mode for the cache. Transactional caches are not shared across nodes and are not replicated. Transactional caches are useful for high-performance transactional data stores.",
+          "non-xa": "NON_XA",
+          "non-xa-tooltip": "NON_XA tooltip",
```
Comment (pruivo) [inline src/app/assets/languages/en.json:274]:
```suggestion
          "non-xa-tooltip": "{{brandname}} registers itself using Synchronizations",
```

File: src/app/assets/languages/en.json [hunk 6]
```diff
@@ -265,7 +265,21 @@
...
+          "non-durable-xa": "NON_DURABLE_XA",
+          "non-durable-xa-tooltip": "NON_DURABLE_XA tooltip",
```
Comment (pruivo) [inline src/app/assets/languages/en.json:276]:
```suggestion
          "non-durable-xa-tooltip": "{{brandname}} registers itself using Xa Resouce but with recovery disabled.",
```

File: src/app/assets/languages/en.json [hunk 7]
```diff
@@ -265,7 +265,21 @@
...
+          "full-xa": "FULL_XA",
+          "full-xa-tooltip": "FULL_XA tooltip",
```
Comment (pruivo) [inline src/app/assets/languages/en.json:278]:
```suggestion
          "full-xa-tooltip": "{{brandname}} registers itself using Xa Resouce with recovery enabled.",
```

**Comment by oraNod** [comment]
@pruivo I'll do it today. Thanks for the heads up.

**Comment by karesti** [comment]
@dipguptaredhat check the changes I did (renaming + feature selection + alert message)

**Pull request merged**